src/config/settings/io_settings.rs

```diff
@@ -0,0 +1,62 @@
+use crate::config::settings::settings_fields::Settings_fields;
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("File interaction failed : {0}")]
+    FileError(#[from] std::io::Error),
+    #[error("Failed to read json : {0}")]
+    JsonError(String),
```

Comment:
```suggestion
    ParsingError(#[from] serde_json::Error),
```

Reply:
give more business indication for the current module than just the technical problem